### ISSUE TITLE
🐛 [Bug Report] Quest design inconsistencies in Coldridge Valley

### ISSUE THREAD

**Issue by SeelGit** [cmangos/issues#3753]
### Bug Details

While making an equivalent playthroughs in Cmangos TBC and vanilla Classic I I discovered several problems with inconsistent quest design in starter zone for dwarves. All the points are relevant against the tbc-db if not mentioned otherwise.

Quest design changes:
1. In Cmangos an NPC [Sten Stoutarm - 658](https://www.wowhead.com/classic/npc=658/sten-stoutarm) can give two quests to new player, [Dwarven Outfitters - 179](https://www.wowhead.com/classic/quest=179/dwarven-outfitters) and [Coldridge Valley Mail Delivery (p.1) - 233](https://www.wowhead.com/classic/quest=233/coldridge-valley-mail-delivery). In Classic only 179 is initially available and when you complete it - that's when 233 becomes available. Wowhead seems to be wrong about this, as it doesn't mention anything about the prerequisite for 233, but in comments there two different players are telling that they also couldn't take 233 before finishing 179. It also seems very natural to have this progression, as 179 requires killing mobs right at the spawn point for new players and 233 leads them in other direction, where other starter quests are happening, so opening it after the end of first quest is kind of nice. Note: wotlk-db doesn't have this problem as checked by @Grz3s.
~~2. [The Troll Cave - 182](https://www.wowhead.com/classic/quest=182/the-troll-cave) rewards also contains additional reward in the form of 3 x [Healing Herb - 961](https://www.wowhead.com/classic/item=961/healing-herb), this is not included in tbc-db, only main 4 items are available to be chosen from.~~
3. [Supplies to Tannok - 2160](https://www.wowhead.com/classic/quest=2160/supplies-to-tannok) is immediately available in cmangos to a new 1st level player. In Classic it only became available for me when I took [Senir's Observations - 420](https://www.wowhead.com/classic/quest=420/senirs-observations), in itself part of a chain of 4 quests. UPD: it's more correct to say that [Senir's Observations - 282](https://www.wowhead.com/classic/quest=282/senirs-observations), previous quest in the chain needs to be completed before taking 2160.
This is very expected, as 2160 leads you to the same location as 420 and outside of Coldridge Valley.
~~4. Maybe I'm misunderstanding something, but it looks like wotlk-db is missing RewRepValue1 and RewRepValue2 for all the quests mentioned in this issue (179, 233, 420, 2160, 3364, maybe others in this area?)~~

Cosmetic quest changes:
5. At the end of [Scalding Mornbrew Delivery - 3364](https://www.wowhead.com/classic/quest=3364/scalding-mornbrew-delivery), [Durnan Furcutter - 836](https://www.wowhead.com/classic/npc=836/durnan-furcutter), the receiver of the delivery, is saying two quotes (listed on wowhead) and dances. In cmangos he doesn't dance.
6. Some English quest descriptions and text are inconsistent between tbc and wotlk-db, will be easier to list in the actual PR for this issue.

World content / loot changes:
7. In troll cave, namesake of the quest 182 mentioned above, there is a chest spawning at a certain part of the cave, containing some coins, red apple(4536) and a cloak(2652), that I haven't encountered in cmangos at the same location.
8. One of the [Frostmane Troll Whelp - 706](https://www.wowhead.com/classic/npc=706/frostmane-troll-whelp) is stuck in fallen tree next to to the troll cave. Can be agroed out of it, but still positions seems to be wrong.

RU localization problems are also abound in Coldridge Valley, from missing text to wrong text assigned to NPCs, to missing gendered text variations, wrong number of mob kills needed to finish the quest in description of the quest. I plan to separate this to another PR and issue to not mix changes to quest design, content and English text and to localization files.

### Steps to Reproduce

All the content of the issue is available in starter zone for the race while leveling 1-6.

### Suggested Workaround

I would have liked to fix all of the aforementioned issued myself.
I'm absolutely sure, verified with the official Classic server and have screenshots (unfortunately in Ru locale, as that is what I'm working on in the background) for all the points other than 6. For 6 I have a screenshot of contents of the chest, and independent comment on wowhead talking about it, but I don't see it logged in sniff I captured while playing and don't yet know how to check if it actually exists in tbc-db, maybe I was just unlucky with the spawn of the chest? 

I would like to get some directions from maintainers if possible:
a) All the problems are related to vanilla content, not tbc, but I'm playing on cmangos-tbc and looking at the content of it's database. I started to compare quest_template entries between wotlk-db and tbc-db, and I can probably do the same for classic-db. Question is - against which DB do I need to make the changes to mostly guarantee that they will also reach tbc-db that I care the most about right now? I seen in accepted PRs in tbc-db that many commits were actually cherrypicked from classic-db changes. Creating manual update .sql files and sending separate PRs to all three dbs seems like a very tedious approach. What are the preferences and expectations of maintainer team?
b) Following the first quest - is there any sense in normalizing the annoying disrepencies between tbc and wotlk db in swapped RewRepFaction1 and RewRepFaction2? It's not functionally an error (except for wotlk also missing actual values for the rewards) but it also makes automatic comparison of the quest rows between dbs more complicated.
c) For problem 5 - can you please give an example where to look to understand how to make NPC play animations in similar situations?
d) For problem 7 and 8 - to be completely sure what to change, it will really help to know how to get some kind of object id in-game to cross-check between official server and cmangos dbs. Is this even possible for chests and random mobs? Can you recommend something like a plugin for modern 1.15.* clients to do this? Will Wowhead Client allow this for chest and normal mob?
e) If I understand correctly - expected format for the changes is a few separate sql "update" files, not updates to the base Full DB file and not an a change in the existing files in "Updates" dir?

### Core SHA1 Commit Hash

155217c690b9dbddc0615f309248429adf074fb7

### Database SHA1 Commit Hash

a56aff0dafe006242cdc54b24ebae7337e6e5e6d

### Operating System

Win 11

### Client Version

2.4.3 (The Burning Crusade)

**Comment by SeelGit** [comment]
First pack of screenshots comparing content between cmangos tbc and classic (sorry again for Ru loc, I think you will still be able to see what is going on):
2)
![2](https://github.com/user-attachments/assets/39630048-2a72-423b-af45-06ea3341c1ce)
3)
![3](https://github.com/user-attachments/assets/ec2ddf79-a4ef-4659-8f86-bb49e8bb2040)
7)
![7](https://github.com/user-attachments/assets/d0b210cc-fdd7-4c75-9caa-a49a9122ce01)
8)
![8](https://github.com/user-attachments/assets/1adaab2a-9632-4eb8-bb16-d27959550f5f)

**Comment by SeelGit** [comment]
Also, I'm a bit lost about the issue 2 now (missing 3 x Healing Herb - 961 reward). It exists in my local cmangos-tbc DB, but in game it's not shown, as evident from the screenshot.
What can be a reason for this I don't know. Will probably require independent repro.

**Comment by SeelGit** [comment]
Issue 2 is caused by playing on 2.5.3 client through HermesProxy, no repro on 2.4.3, works correctly there, not a cmangos problem. Removing 2 from the list.

**Comment by SeelGit** [comment]
```sql
-- ID 233 Coldridge Valley Mail Delivery (p.1) - opens only when the intro quest is finished (ID 179 Dwarven Outfitters) 
UPDATE quest_template SET PrevQuestId = 179 WHERE entry = 233;

-- ID 2160 Supplies to Tannok - ID 282 Senir's Observations chain leads player out of Coldridge Vallet, you need to agree to it before taking 2160
UPDATE quest_template SET PrevQuestId = 282 WHERE entry = 2160;
```
Fix for 1 and 2.

**Comment by celguar** [comment]
next time please use 2.4.3 for all your tests and pics. It might be ok but would be a waste of time if some issues are caused by an imperfect Hermes Proxy + 2.5.3 and devs check it in vain

**Comment by celguar** [comment]
Stuck in tree might be a mmap issue, is not a trivial task to fix. I have a fix on my fork, but is not perfect either

**Comment by SeelGit** [comment]
> next time please use 2.4.3 for all your tests and pics

sure, already started using exclusively 2.4.3 client for all future tests and gameplay. Mentioned quest issues are relevant for 2.4.3, missing chest in the cave is also not appearing when playing with 2.4.3 client.

**Comment by celguar** [comment]
"wrong number of mob kills needed to finish the quest in description of the quest." what quest has it?

**Comment by celguar** [comment]
> missing chest in the cave is also not appearing when playing with 2.4.3 client.

just started 2.4.3 cmangos, chest is there

![Chest](https://cdn.discordapp.com/attachments/836430388779548682/1274840457578287238/image.png?ex=66c3b732&is=66c265b2&hm=fdb164c44983fd9c1230efa9d3df0f04c392f3246f99eff4cb0ee365711c738f&)

**Comment by celguar** [comment]
or it was different chest? How many are there? 🔢

**Comment by SeelGit** [comment]
Yeah, this is a fluke. Instead of saying to you that I played through this cave with 3 identical characters and never got the chest, 
 I tried once more and with my fourth dwarf test char, I got chest to spawn. 
I was incorrectly assuming that if I was playing solo with bots and the cave haven't been visibly cleared before me going there, this should guarantee the chest spawn, but that evidently is not the case.
I also now understand a bit better how the game works so I will check if I can get object ids to display for future similar bug reports and cross reference that id with the DB entries.
Sorry, removing chest from the list.

**Comment by SeelGit** [comment]
I guess 4 is irrelevant and WoTLK server assigns rep reward values automatically based on quest level, without needing explicit value, so I will also remove that one if there is no evidence that this is an actual error. 

For 5 I will recheck again with a new playthrough on 2.4.3 to understand if emote is actually there. If it's not, I will record a video proof.

**Comment by SeelGit** [comment]
> "wrong number of mob kills needed to finish the quest in description of the quest." what quest has it?

Note, this is only a problem in localization text, actual values in quest template are correct. 
Quests 182 and 183 both list incorrect number of trolls and boars to kill to complete quest in initial quest text in Russian localization. I already have SQL files ready with Russian translation fixes for all the quests in Coldridge Valley, will provide them in a separate issue and PR to not mix localization and content fixes here.

**Comment by celguar** [comment]
Well chest spawn you linked from classic I did not see, the one I saw in cmangos tbc was very close but on lower level.

**Comment by SeelGit** [comment]
You are right, it's a different location on your screenshot. And when testing again in cmangos I then got chest, again, with very similar contents but at a third placement point, 
![image](https://github.com/user-attachments/assets/1f31e6c4-d6cc-4356-b81a-83395cb60840)
different from both your screen and classic screen, so there are multiple chest locations in this cave at least in cmangos and maybe in vanilla too?

But returning to your screenshot, it looks pretty similar, but on a lower level. I'm not sure yet how to check if this is another valid chest spawn point or if the chest spawn point in cmangos is just wrong on Z coordinate because, for example, it was added to the game based just on roughly known x and y coordinates.
Here is comparison of your map and my classic screenshot
![sdsdf](https://github.com/user-attachments/assets/f8ac2878-788c-4265-94a8-da66f2fb55b3)
